fix: clarify documentation for AutofacChildLifetimeScopeServiceProviderFactory (#92)
autofac/Documentation: docs/integration/aspnetcore.rst

    public class Program
    {
      public static async Task Main(string[] args)
      {
        // create the root-container and register global dependencies
        var containerBuilder = new ContainerBuilder();

        builder.RegisterType<SomeGlobalDependency>()
          .As<ISomeGlobalDependency>()
          .InstancePerLifetimeScope();

        var container = containerBuilder.Build();

        // The UseServiceProviderFactory call attaches the
        // Autofac provider to the generic hosting mechanism.
          var hostOne = Host
            .CreateDefaultBuilder(args)

>             .UseServiceProviderFactory(new AutofacChildLifetimeScopeServiceProviderFactory(container))

Reviewer: ```suggestion
            .UseServiceProviderFactory(new AutofacChildLifetimeScopeServiceProviderFactory(container.BeginLifetimeScope("root-one")))
```

The idea here is that the "root" can be a child scope of the container, it doesn't have to be a container at all. I'll update the two examples with some suggestions and we should be able to call it good.

            .ConfigureWebHostDefaults(webHostBuilder => {
              webHostBuilder
                .UseContentRoot(AppContext.BaseDirectory)

>                 // each host listens to a different URL while still using the same root-container

Reviewer: ```suggestion
                // Each host listens to a different URL, they have the same root container to share SingleInstance
                // things, but they each have  their own logical root lifetime scope. Registering things
                // as `InstancePerMatchingLifetimeScope("root-one")` (the name of the scope given above)
                // will result in a singleton that's ONLY used by this first host.
```

                .UseUrls("http://localhost:5000")
                .UseStartup<StartupOne>();
            })
            .Build();

        // The UseServiceProviderFactory call attaches the
        // Autofac provider to the generic hosting mechanism.
          var hostTwo = Host
            .CreateDefaultBuilder(args)

>             .UseServiceProviderFactory(new AutofacChildLifetimeScopeServiceProviderFactory(container))

Reviewer: ```suggestion
            .UseServiceProviderFactory(new AutofacChildLifetimeScopeServiceProviderFactory(container.BeginLifetimeScope("root-two")))
```

            .ConfigureWebHostDefaults(webHostBuilder => {
              webHostBuilder
                .UseContentRoot(AppContext.BaseDirectory)

>                 // each host listens to a different URL while still using the same root-container

Reviewer: ```suggestion
                // As with the first host, the second host will share the root container but have its own
                // root lifetime scope `root-two`. Things registered `InstancePerMatchingLifetimeScope("root-two")`
                // will be singletons ONLY used by this second host.
```
